static/src/js/components/CustomSelectWidget/CustomSelectWidget.jsx

```diff
@@ -158,7 +158,7 @@ const CustomSelectWidget = ({
   const { enumOptions: oneOfEnums } = options || {}
 
   const getExistingValue = () => {
-    if (value) {
+    if (value || value >= 0) {
       if (typeof value === 'number') {
```

Comment:
do you need line 161, if so, maybe combine, `if (value && typeof value === 'number') {`

Reply:
I tried that but no luck. This did not work in the case where the value is a string. 


```
    if (value !== null) {
      if (typeof value === 'number') {
        return oneOfEnums[value]
      }

      return {
        value,
        label: value
      }
    }

    return null
  }
```

the snippet above worked as expected and covered all of the cases